src/useRefs/useRefs.ts

```diff
@@ -0,0 +1,43 @@
+import { createRef, useMemo, type RefObject } from 'react';
+
+/**
+ * Utility to automatically create refs
+ *
+ * ```typescript
+ * const buttonRef = useRef<Array<HTMLButtonElement>>([]);
+ *
+ * const refs = useRef<{
+ *  heading: RefObject<HTMLHeadingElement | null>
+ *  button: RefObject<Array<HTMLButtonElement | null>>
+ * }>({
+ *   buttonRef
+ * })
+ * ```
+ *
+ * refs.heading -> ref automatically created
+ * refs.button -> ref from target object
+ */
+export function useRefs<T extends Record<string | symbol, RefObject<unknown>>>(
+  initialTarget?: Record<string | symbol, RefObject<unknown>>,
```

Comment:
The MDX documentation still says `initialTarget?: Partial<T>,`, which I see is not how this is typed. Must an oversight on the implementation?

Also, the JS docs above allows passing `buttonRef` while the generic expects `button`. That should probably be updated as well.

Reply:
Updated the type to `Partial<T>`